base/initdefs.jl

```diff
@@ -40,8 +40,13 @@ function init_load_path()
     if haskey(ENV, "JULIA_LOAD_PATH")
         prepend!(LOAD_PATH, split(ENV["JULIA_LOAD_PATH"], @static is_windows() ? ';' : ':'))
     end
-    push!(LOAD_PATH, abspath(JULIA_HOME, "..", "local", "share", "julia", "site", vers))
-    push!(LOAD_PATH, abspath(JULIA_HOME, "..", "share", "julia", "site", vers))
+    if Base.DARWIN_FRAMEWORK 
+        push!(LOAD_PATH, abspath(homedir(),"Library",Base.DARWIN_FRAMEWORK_NAME,vers,"site"))
+        push!(LOAD_PATH, abspath("/Library",Base.DARWIN_FRAMEWORK_NAME,vers,"site"))
```

Comment:
note the comment at the top of `Makefile`
> Third-party code that is not controlled by Pkg should be installed into prefix/share/julia/site/VERSDIR (not prefix/share/julia/VERSDIR/site ... so that prefix/share/julia/VERSDIR can be overwritten without touching third-party code).